game/hud/src/widgets/Crafting/components/App/index.tsx

```diff
@@ -158,13 +157,13 @@ class App extends React.Component<AppProps,AppState> {
             setQuality={this.setQuality}
             setCount={this.setCount}
             setRecipe={this.setRecipe}
-            setName={this.setName} setTemplate={this.setTemplate}
+            setName={this.setName}
             addIngredient={this.addIngredient}
             removeIngredient={this.removeIngredient}
           />
         );
       case 'tools':
-        return <Tools dispatch={this.props.dispatch} />;
+        return <Tools dispatch={this.props.dispatch} refresh={() => this.refresh()}/>;
```

Comment:
Two things.
1.) You don't actually need to pass dispatch down to the Tools component because it is using connect(). connect() should already provide dispatch() as a prop.

2.) You can just use `refresh={this.refresh}` instead of `refresh={() => this.refresh()}`

Reply:
Humm... Except that, to make Tools compile, we made dispatch: not optional, so App doesn't compile if don't pass dispatch.

Reply:
If you don't pass it down then I think you'd have to include dispatch into the ToolsReduxProps instead because it is now being provided by connect().

Reply:
That didn't work either, but thinking about the error message I was getting, which basically said, `typeof Tools != StatelessComponent<ToolsOwnProps & DispatchProp<any> & ToolsPropsRedux>` , I tried this, and it works, and what's more, makes sense.

```
import { connect, DispatchProp } from 'react-redux';
export interface ToolsPropsRedux {
  countdown: number;
}
export interface ToolsOwnProps {
  style?: Partial<ToolsStyles>;
  refresh: () => void;
}
const select = (state: GlobalState, props: ToolsOwnProps) : ToolsPropsRedux => {
  return { countdown: state.ui.countdown };
};
export type ToolsProps = ToolsPropsRedux & DispatchProp<any> & ToolsOwnProps;
```
Though why other components which do similar build ok (e.g. `VoxInfo`), IDK /shrug

Reply:
¯\_(ツ)_/¯
We should look into why that is some time and see what's up, could always be a bug in the typings.

Reply:
```
export type ToolsProps = ToolsPropsRedux & DispatchProp<any> & ToolsOwnProps;
```
Is probably more correct anyway.